README.md

```diff
@@ -1,105 +1,146 @@
+    // Just an example, customise as needed.
+    new GuAllowPolicy(stack, "GitHubBucketAccessPolicy", {
+      actions: ["s3:Put*"],
+      resources: ["arn:aws:s3:::/build-artifacts/*"],
+    }),
+  ],
+});
 ```
 
-See the [toolkit documentation](https://github.com/actions/toolkit/blob/master/README.md#packages) for the various packages.
-
-## Publish to a distribution branch
-
-Actions are run from GitHub repos so we will checkin the packed dist folder. 
+If you're using writing CloudFormation in YAML, the resources you need will look something like this:
 
-Then run [ncc](https://github.com/zeit/ncc) and push the results:
-```bash
-$ npm run package
-$ git add dist
-$ git commit -a -m "prod dependencies"
-$ git push origin releases/v1
+```yaml
+Resources:
+  # Resources to provide S3 upload permissions to GitHub Actions.
+  # Permissions are scoped to Riff-Raff buckets.
+  # See https://github.com/guardian/actions-assume-aws-role
+  GitHubBucketAccessPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: GitHubBucketAccessPolicy
+      PolicyDocument:
+        Statement:
+          Effect: Allow
+          Action:
+            - s3:Put* # Just an example, customise as needed.
+          Resource:
+            - !Sub arn:aws:s3:::build-artifacts/* # Just an example, customise as needed.
+      Roles:
+        - Ref: GitHubRole
+
+  GitHubRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action: sts:AssumeRoleWithWebIdentity
+            Principal:
+              Federated: !Ref GitHubOidc
+            Condition:
+              StringLike:
+                # All GitHub Actions running in repositories within the Guardian GitHub organisation, customise as needed.
+                vstoken.actions.githubusercontent.com:sub: repo:guardian/*
+
+  GitHubOidc:
+    Type: AWS::IAM::OIDCProvider
+    Properties:
+      Url: https://vstoken.actions.githubusercontent.com
+      ClientIdList: [sigstore]
+
+      # This is the thumbprint of `vstoken.actions.githubusercontent.com`
+      # See: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html
+      ThumbprintList: [a031c46782e6e6c662c2c87c76da9aa62ccabd8e]
+
+Outputs:
+  GitHubRoleArn:
+    # To be set as a secret in GitHub Actions
+    Value: !GetAtt GitHubRole.Arn
 ```
 
-Note: We recommend using the `--license` option for ncc, which will create a license file for all of the production node modules used in your project.
-
-Your action is now published! :rocket: 
-
-See the [versioning documentation](https://github.com/actions/toolkit/blob/master/docs/action-versioning.md)
-
-## Validate
-
-You can now validate the action by referencing `./` in a workflow in your repo (see [test.yml](.github/workflows/test.yml))
+## Usage
+1. Add the `AWS::IAM::Role` ARN as a GitHub Actions [secret] (we'll assume the secret has been called `GU_ACTIONS_ROLE`)
```

Comment:
I think steps 1 & 2 might be taken care of for Guardian projects? If so, perhaps we can make another note about that?